modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/GranuleCatalog.java

```diff
@@ -147,4 +151,12 @@ protected MultiLevelROI getGranuleFootprint(SimpleFeature sf) {
      * @throws IOException in case something bad happens
      */
     public abstract void drop() throws IOException;
+
+    public SubmosaicProducerFactory getCollectorsFactory() {
```

Comment:
It is not the job of the granule catalog to know about the collectors/mosaic producers, it is just to know about the granule originators and descriptors, but not about how to build the results. The raster manager should pick it from the MosaicConfigurationBean on initialization instead, and probably make it a field to avoid re-instantiating it over and over.